### DISCUSSION TITLE
Ebyte E104-BT11 modules based on EFR32MG21 used with Zigbee firmware?

### DISCUSSION THREAD

**Discussion by Hedda** [zigpy/zigpy#727] [General]
@grobasoz Do you think the new E104-BT11G modules from Ebyte are based on EFR32MG21 so could be re-flashed with Zigbee firmware?

https://www.cdebyte.net/bluetooth-module/2-4g-ble-mesh-network-bluetooth-module.html

https://www.manualslib.com/manual/1895760/Ebyte-E104-Bt11.html

https://www.ebyte.com/en/downpdf.aspx?id=1010

https://www.ebyte.com/downpdf.aspx?id=990

![image](https://user-images.githubusercontent.com/6320001/116984871-06a95a80-accc-11eb-853f-7272a17b2f96.png)

![image](https://user-images.githubusercontent.com/6320001/116984822-f42f2100-accb-11eb-80f0-dd65dc42cf15.png)  

Examples in the series are "E104-BT11G-IPX" (IPX antenna) and Ebyte "E104-BT11G-PCB" (PCB trace antenna):

Name | Remark | Antenna Type | Size | Distance
-- | -- | -- | -- | --
E104-BT11G-IPX | GATEWAY MODULE | IPX Antenna | 16*13mm | 300m
E104-BT11G-PCB | GATEWAY MODULE | PCB Antenna | 19*13mm | 200m
E104-BT11N-IPX | NODE MODULE | IPX Antenna | 16*13mm | 300m
E104-BT11N-PCB | NODE MODULE | PCB Antenna | 19*13mm | 200m

https://www.aliexpress.com/wholesale?catId=0&SearchText=E104-BT11G-PCB

https://www.aliexpress.com/wholesale?catId=0&SearchText=E104-BT11G-IPX

https://www.alibaba.com/product-detail/GFSK-BLE-mesh-integrated-circuits-blue_1600200894074.html

https://www.alibaba.com/product-detail/E104-BT11G-PCB-EFR32-2-4g_1600080880779.html

![image](https://user-images.githubusercontent.com/6320001/116983510-7f0f1c00-acca-11eb-90f2-caabc1ee63b8.png)

## Pin definition

Pin | Function | Description 
-- | -- | -- | -- 
1 | GND | Ground, connect to power reference ground
13 | NetLink | Network indicator, pin outputs high level after successfully networking
17 | TXD | TX pin of UART，for UART receiving
18 | RDX | RX pin of UART，for UART transmitting
19 | VCC | Power supply 1.9～3.6V（It is recommended to add a ceramic filter capacitor externally)
20 | GND | Ground, connect to power reference ground
22 | PC03 | Low-power role selection pin (internal pull-up). Low power consumption role: pull up or float. Other roles: Go directly to the ground.
23 | PC04 | Reset. Except for hardware information (UART configuration), factory settings are restored.
24 | Wakeup | The wake-up pin is valid in low level under low power mode.
25 | RST | Chip reset trigger input pin, valid in low level
26 | GND | Ground, connect to power reference ground

![image](https://user-images.githubusercontent.com/6320001/116985036-3fe1ca80-accc-11eb-91e4-aebda3383dae.png)

**Comment by MattWestb** [comment]
With one j-tag and doing chip erase all shall being possible if not doing the chip 110% debug locked = then its one hard bricked device.

Pad 10 SWD 11 SWC 19 VCC 20 GND connected and reading the chip id  with simplicity commander and you is knowing the real chip.

Normally EByte is shipping without bootloader but they can have changing that in the new products  but i can being that they have the new combined bootloader.

The G-version is configured as one "coordinator" and the N as router device all with BT mesh.

**Reply by grobasoz** [reply]
Sounds about right :)

**Reply by Hedda** [reply]
So it will probably b similar to the E180-Z120B module E180-ZG120B-TB board by Ebyte which @tube0013 started out using?

https://github.com/zigpy/zigpy/discussions/584

https://www.cnx-software.com/2020/12/21/diy-project-creates-zigbee-to-ethernet-bridge-with-wt32-eth01/

https://www.cnx-software.com/2020/04/27/ebyte-e180-zg120b-tb-zigbee-3-0-evaluation-board-features-silicon-labs-efr32mg1b-zigbee-thread-soc/

https://github.com/zha-ng/EZSP-Firmware

https://github.com/zha-ng/EZSP-Firmware/tree/master/EByte-E180-Z120B

https://github.com/zha-ng/EZSP-Firmware/issues/2

# EByte-E180-Z120B

**EByte-E180-Z120B** comes with a proprietary firmware. To update it with an EZSP based firmware you have to use a SWD Flasher at least once to upload the bootloader.
When uploading the bootloader for the very 1st time, use the _combined.s37_ image.
How to program the bootloader is out of the scope, however how-to updates are welcome. See https://github.com/myelin/arduino-cmsis-dap and [Open On-Chip Debugger](http://openocd.org/)

## efr32mg1b-bootloader
- EFR32MG1B target
- NCP UART TX --> PA0
- NCP UART RC <-- PA1
- PB12/UART_BUTTON_RESET is bootloader activiation pin/button
- Version 1.10.3
- DCDC

## efr32mg1b-v8-678-57600
- EFR32MG1B target
- NCP UART TX --> PA0
- NCP UART RC <-- PA1
- EZSP Version 8
- EmberZNet 6.7.8
- DCDC

## efr32mg1b-v8-6910-57600
- EFR32MG1B target
- NCP UART TX --> PA0
- NCP UART RC <-- PA1
- EZSP Version 8
- EmberZNet 6.9.1.0
- DCDC

## efr32mg1b-v8-6910-115200
- EFR32MG1B target
- NCP UART TX --> PA0
- NCP UART RC <-- PA1
- EZSP Version 8
- EmberZNet 6.9.1.0
- DCDC

## efr32mg1b-latest-57600
- EmberZNet 6.8.2.0
- Everything else is same as above

## efr32mg1b-latest-115200
- EmberZNet 6.8.2.0
- Everything else is same as above

Configuration Parameter | Value
----------------------- | ------
Address Table Size | 16
Child Table Size | 32
Source Routes | 200

The remaining parameters are at the default values.

**Reply by tube0013** [reply]
Question is which chip they are using and ram/flash available

**Reply by MattWestb** [reply]
Is one 20 dBm chip and very likely one a series so 3 different variants:
512 / 64,  768 / 64 and 1024 / 96 (and the same for b series).

I have not finding any FCC cert but looks better then the unshielded SM-011 that is having much problems with interference in the module.

I have running one tuya / LIDL LED stripe controller with one EFR32MG21 and EZSP 6.9.2.0 and its looks being very stable and dont having problems with acking and replaying to its children that other coordinators is having.

**Reply by MattWestb** [reply]
Is the `arduino-cmsis-dap` verified working on MG2X chips or is it only working on first gen chips ??
And what application using for doing the flashing of the chip with the `cmsis` adapter ?

I have only using BMP (Black Magic Probe) with GDB with MG1X chips and also Silabs WSTK for MG2X chips.

**Reply by MattWestb** [reply]
> Wireless MCU – Silicon Labs EFR32MG21 Arm Cortex-M33 microcontroller @ up to 80 MHz with DSP and FPU, up to 1024 kB flash, up to 96 kB RAM, 2.4 GHz IEEE 802.15.4 for Bluetooth LE, Zigbee, or Thread

They is only saying "up to" = not that it is the maximum but one MG21 A or B  version is OK for My :-))
